marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.hub/lib/perflog-lib.xqy

```diff
@@ -0,0 +1,33 @@
+(:
+  Copyright 2012-2016 MarkLogic Corporation
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+:)
+xquery version "1.0-ml";
+
+module namespace perflog = "http://marklogic.com/data-hub/perflog-lib";
+
+(:~
+ : determine if a module exists or not
+ :
+ : @param $uri - the uri of the module to check
+ : @return - true if exists, false otherwise
+ :)
+declare function perflog:logit($what-is-it, $x)
```

Comment:
You weren't on the call when I talked about this code. I mentioned that it was an example and that you all shouldn't use the exact names as I wrote it in about 45 seconds. I'd like to see the variable names make a little more sense.

Also I'd prefer to see some sort of greppable thing in the log message. Maybe something like:

xdmp:log("PERFORMANCE: " || $what-is-it || " took " || $time)

You should also look at the log file output to make sure it makes sense. I'm pretty sure $time is something like "PT0.00001S" and doesn't need the ms at the end.